lib/trino-orc/src/main/java/io/trino/orc/OrcWriter.java

```diff
@@ -511,6 +512,9 @@ private List<OrcDataOutput> bufferFileFooter()
         fileStatsRetainedBytes = fileStats.map(stats -> stats.stream()
                 .mapToLong(ColumnStatistics::getRetainedSizeInBytes)
                 .sum()).orElse(0L);
+        stripeOffsets = closedStripes.stream()
+                .map(closedStripe -> closedStripe.getStripeInformation().getOffset())
```

Comment:
It means that we get 1 split per parquet row-group or orc stripe instead of 128MB (or configured property) size split which might be mapped to 0-N row-groups depending on row group size configured in the writer. It should give us same or more parallelism and avoid any empty splits due to mis-alignment of logical split offsets with file row-group boundaries.
We were already getting the same behaviour when the data was written by some other engine/library which would populate split offsets.

Reply:
> It means that we get 1 split per parquet row-group or orc stripe instead of 128MB (or configured property) size split which might be mapped to 0-N row-groups depending on row group size configured in the writer.

Do we have minimal split size if the row groups are super tiny, e.g: to avoid reading Parquet footers over and over again for tiny tiny stripes?

Reply:
We rely on iceberg library logic to split the file and it doesn't have such a fallback. There's no good reason why any writer would be producing a lot of small row-groups, it would end up nullifying all the advantage of a columnar file format. Keep in mind that other writers are already populating this field and we've already been relying on the same iceberg logic for data written in this way. If someone has such data due to some misconfiguration, they need to fix it through running OPTIMIZE or CTAS.

Reply:
> There's no good reason why any writer would be producing a lot of small row-groups

Right, but small row-groups could be a byproduct of wide rows (I think I've seen that happening)